packages/cli/src/databases/repositories/workflowStatistics.repository.ts

```diff
@@ -98,4 +98,23 @@ export class WorkflowStatisticsRepository extends Repository<WorkflowStatistics>
 			throw error;
 		}
 	}
+
+	async queryNumWorkflowsUserHasWithOver5ProdExecs(userId: User['id']): Promise<number> {
+		const numWorkflows = await this.createQueryBuilder('workflow_statistics')
```

Comment:
Can we return this directly?

Reply:
I've noticed we use extensively the pattern of returning a promises directly from functions instead of awaiting them (and we even have a eslint rule to forbid await in return). Sure, in that case the function is no longer in the stack and there's no micro task in the event loop, but we lose visibility. Because it's no longer in the stack, it won't be shown in the stack trace either if there ever is an error thrown somewhere. Many [consider that an anti pattern](https://github.com/goldbergyoni/nodebestpractices#-212-always-await-promises-before-returning-to-avoid-a-partial-stacktrace).

So to answer your question, yes, we could return directly, but I'd rather keep it like this.

Reply:
Interesting, I didn't know this! 

To better understand - Do you mean [this lint rule](https://eslint.org/docs/latest/rules/no-return-await)? That's deprecated and we don't seem to be using it? Also if we were to return directly, we wouldn't see `createQueryBuilder` but we would see `queryNumWorkflowsUserHasWithOver5ProdExecs` in the stack trace? So in end effect we would still locate the issue or I'm misunderstanding?

Reply:
It seems to be [this rule](https://typescript-eslint.io/rules/return-await/) from `typescript-eslint`. It would drop the `queryNumWorkflowsUserHasWithOver5ProdExecs` from the stack trace either, since that function is not attached to the promise chain anymore. There are some examples [here](https://github.com/goldbergyoni/nodebestpractices/blob/master/sections/errorhandling/returningpromises.md).

Reply:
Thank you!